src/metrics/prometheus_histogram.erl

```diff
@@ -253,7 +255,7 @@ observe_n(Registry, Name, LabelValues, Value, Count) when is_integer(Value), is_
             insert_metric(Registry, Name, LabelValues, Value, Count, fun observe_n/5)
     end,
     ok;
-observe_n(Registry, Name, LabelValues, Value, Count) when is_number(Value), is_integer(Count) ->
+observe_n(Registry, Name, LabelValues, Value, Count) when is_float(Value), is_integer(Count) ->
```

Comment:
This looks odd in isolation, until one sees the `is_integer/1` guard on the previous clause.